src/networking.c

```diff
@@ -2353,6 +2354,12 @@ NULL
     } else if (!strcasecmp(c->argv[1]->ptr,"id") && c->argc == 2) {
         /* CLIENT ID */
         addReplyLongLong(c,c->id);
+    } else if (!strcasecmp(c->argv[1]->ptr,"info")) {
+        /* CLIENT INFO */
+        sds o = catClientInfoString(sdsempty(), c);
+        o = sdscatlen(o,"\n",1);
```

Comment:
are you sure that newline is needed? did you add it for the benefit of redis-cli?
or maybe redis-cli needs special handling for that (same as it does for CLIENT LIST).

Reply:
Both are true, `redis-cli` does need special handling to display this the same as `CLIENT LIST`. As for other clients, the newline is there to make parsing identical to `CLIENT LIST` which was the original intent.

I believe this represents a tradeoff between better structuring of the data vs. compatibility with existing `CLIENT LIST` parsers.

Reply:
@yossigo are you gonna fix redis-cli?